# Аналіз набору даних Students Social Media Addiction

## Метою первинного аналізу є перевірка якості даних та оцінка їх придатності для подальшого аналізу і візуалізації.

In [1]:
import pandas as pd
import numpy as np

# 1. Завантаження датасету
df = pd.read_csv('Students Social Media Addiction.csv')

# 2. Попередній перегляд перших 5 рядків
print("Перші 5 записів датасету:")
display(df.head())

# 3. Перевірка структури даних та типів колонок
print("\nІнформація про типи даних та пропуски:")
df.info()

# 4. Перевірка на наявність дублікатів
duplicates = df.duplicated().sum()
print(f"\nКількість дублікатів у даних: {duplicates}")

# 5. Перевірка на пропуски (null values)
missing_values = df.isnull().sum()
print("\nКількість пропущених значень у кожній колонці:")
print(missing_values)

# 6. Опис числових даних (мінімум, максимум, середнє)
print("\nСтатистичний опис числових колонок:")
display(df.describe())

Перші 5 записів датасету:


Student_ID  Age  ... Conflicts_Over_Social_Media Addicted_Score
0           1   19  ...                           3              8
1           2   22  ...                           0              3
2           3   20  ...                           4              9
3           4   18  ...                           1              4
4           5   21  ...                           2              7

[5 rows x 13 columns]


Інформація про типи даних та пропуски:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 705 entries, 0 to 704
Data columns (total 13 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Student_ID                    705 non-null    int64  
 1   Age                           705 non-null    int64  
 2   Gender                        705 non-null    object 
 3   Academic_Level                705 non-null    object 
 4   Country                       705 non-null    object 
 5   Avg_Daily_Usage_Hours         705 non-null    float64
 6   Most_Used_Platform            705 non-null    object 
 7   Affects_Academic_Performance  705 non-null    object 
 8   Sleep_Hours_Per_Night         705 non-null    float64
 9   Mental_Health_Score           705 non-null    int64  
 10  Relationship_Status           705 non-null    object 
 11  Conflicts_Over_Social_Media   705 non-null    int64  
 12  Addicted_Score          

Student_ID         Age  ...  Conflicts_Over_Social_Media  Addicted_Score
count  705.000000  705.000000  ...                   705.000000      705.000000
mean   353.000000   20.659574  ...                     2.849645        6.436879
std    203.660256    1.399217  ...                     0.957968        1.587165
min      1.000000   18.000000  ...                     0.000000        2.000000
25%    177.000000   19.000000  ...                     2.000000        5.000000
50%    353.000000   21.000000  ...                     3.000000        7.000000
75%    529.000000   22.000000  ...                     4.000000        8.000000
max    705.000000   24.000000  ...                     5.000000        9.000000

[8 rows x 7 columns]

**Висновки до Блоку 1:**

Якість даних: Висока. Пропуски та дублікати відсутні.

Обсяг: 705 респондентів — це достатня вибірка для статистично значущих висновків.

Необхідність трансформації: Стовпчик Affects_Academic_Performance має текстовий формат (Yes/No), що не дозволить включити його в математичний кореляційний аналіз. Потрібно перевести його в числовий формат (1/0).

**Блок 2: Очищення та трансформація даних**
- Створюємо копію даних.
- Конвертуємо вплив на навчання у цифри (Affects_Academic_Performance у числовий тип).
- Створюємо нову колонку Addiction_Level (Рівень залежності) на основі Addicted_Score для зручності майбутньої кластеризації та групування.
- Перевіряємо структуру оновленого набору даних через .info().
- Зберігаємо очищений датасет у новий файл.

Для чого: Щоб підготувати дані до візуалізації та побудови моделей машинного навчання.

In [2]:
# 1. Створюємо копію датасету, щоб оригінал залишався незмінним
df_cleaned = df.copy()

# 2. Трансформація Affects_Academic_Performance у числовий формат
# 'Yes' перетворюємо на 1, 'No' — на 0
df_cleaned['Affects_Academic_Performance_Numeric'] = df_cleaned['Affects_Academic_Performance'].map({'Yes': 1, 'No': 0})

# 3. Створення категоріальної змінної Addiction_Level
# Розділимо оцінку від 1 до 10 на три рівні:
# 1-3: Low (Низький), 4-7: Medium (Середній), 8-10: High (Високий)
def categorize_addiction(score):
    if score <= 3:
        return 'Low'
    elif score <= 7:
        return 'Medium'
    else:
        return 'High'

# Застосовуємо функцію до колонки Addicted_Score
df_cleaned['Addiction_Level'] = df_cleaned['Addicted_Score'].apply(categorize_addiction)

# 4. Перевірка структури оновленого датасету (ваша пропозиція)
print("Інформація про очищений датасет:")
df_cleaned.info()

# 5. Додатково глянемо на перші кілька рядків, щоб побачити нові колонки в кінці
print("\nПерші 5 рядків очищених даних:")
display(df_cleaned.head())

# 6. Збереження очищеного датасету у новий CSV файл для подальшої роботи
df_cleaned.to_csv('cleaned_social_media_data.csv', index=False)
print("\nФайл 'cleaned_social_media_data.csv' успішно створено та збережено!")

Інформація про очищений датасет:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 705 entries, 0 to 704
Data columns (total 15 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   Student_ID                            705 non-null    int64  
 1   Age                                   705 non-null    int64  
 2   Gender                                705 non-null    object 
 3   Academic_Level                        705 non-null    object 
 4   Country                               705 non-null    object 
 5   Avg_Daily_Usage_Hours                 705 non-null    float64
 6   Most_Used_Platform                    705 non-null    object 
 7   Affects_Academic_Performance          705 non-null    object 
 8   Sleep_Hours_Per_Night                 705 non-null    float64
 9   Mental_Health_Score                   705 non-null    int64  
 10  Relationship_Status                   705 non-null   

Student_ID  Age  ... Affects_Academic_Performance_Numeric Addiction_Level
0           1   19  ...                                    1            High
1           2   22  ...                                    0             Low
2           3   20  ...                                    1            High
3           4   18  ...                                    0          Medium
4           5   21  ...                                    1          Medium

[5 rows x 15 columns]


Файл 'cleaned_social_media_data.csv' успішно створено та збережено!


**Висновки до Блоку 2:**
- Готовність до аналізу: Тепер ми маємо числове представлення впливу на навчання, що дозволить нам побачити, наприклад, чи зростає ймовірність проблем з навчанням при збільшенні годин у соцмережах.

- Сегментація: Нова колонка Addiction_Level дозволить нам будувати красиві порівняльні діаграми (наприклад, "Рівень сну залежно від рівня залежності").

- Кількість колонок: тепер їх 15 (було 13 + 2 нові).

- Типи даних: Affects_Academic_Performance_Numeric - int64, а Addiction_Level — object.

**Блок 3: Кореляційний аналіз та перевірка гіпотез (Гіпотези 1 та 2)**
Що ми робимо:
- Будуємо матрицю кореляції для всіх числових показників.
- Обчислюємо конкретні коефіцієнти для Гіпотези 1 (Зв'язок використання та сну) та Гіпотези 2 (Зв'язок залежності та психічного здоров'я).
- Групуємо дані за Addiction_Level, щоб побачити різницю в "якості життя" між групами.

Для чого: Щоб мовою цифр підтвердити наші припущення. Наприклад, якщо кореляція між використанням та сном буде від'ємною (наприклад, -0.6), ми зможемо математично стверджувати: "Зі зростанням часу в соцмережах якість сну суттєво падає".

In [3]:
# 1. Вибираємо числові колонки для аналізу кореляцій
# Включаємо Age, Usage, Sleep, Mental Health, Conflicts, Addiction Score та нашу нову числову успішність
numerical_cols = [
    'Age', 'Avg_Daily_Usage_Hours', 'Sleep_Hours_Per_Night', 
    'Mental_Health_Score', 'Conflicts_Over_Social_Media', 
    'Addicted_Score', 'Affects_Academic_Performance_Numeric'
]

# Створюємо матрицю кореляції
corr_matrix = df_cleaned[numerical_cols].corr()

print("Матриця кореляції (вплив факторів один на одного):")
display(corr_matrix)

# 2. Перевірка Гіпотези 1: Чи зменшується сон при збільшенні використання?
hypo1_corr = df_cleaned['Avg_Daily_Usage_Hours'].corr(df_cleaned['Sleep_Hours_Per_Night'])
print(f"\nКореляція (Використання vs Сон): {hypo1_corr:.2f}")

# 3. Перевірка Гіпотези 2: Чи страждає психічне здоров'я при зростанні залежності?
hypo2_corr = df_cleaned['Addicted_Score'].corr(df_cleaned['Mental_Health_Score'])
print(f"Кореляція (Залежність vs Психічне здоров'я): {hypo2_corr:.2f}")

# 4. Порівняння груп за рівнем залежності (Addiction_Level)
# Подивимось на середній сон та психічне здоров'я для категорій Low, Medium, High
group_comparison = df_cleaned.groupby('Addiction_Level')[['Sleep_Hours_Per_Night', 'Mental_Health_Score', 'Avg_Daily_Usage_Hours']].mean()

print("\nПорівняння середніх показників за рівнями залежності:")
display(group_comparison)

Матриця кореляції (вплив факторів один на одного):


Age  ...  Affects_Academic_Performance_Numeric
Age                                   1.000000  ...                             -0.137140
Avg_Daily_Usage_Hours                -0.113682  ...                              0.661474
Sleep_Hours_Per_Night                 0.125265  ...                             -0.625373
Mental_Health_Score                   0.160278  ...                             -0.808921
Conflicts_Over_Social_Media          -0.184482  ...                              0.832030
Addicted_Score                       -0.166396  ...                              0.866049
Affects_Academic_Performance_Numeric -0.137140  ...                              1.000000

[7 rows x 7 columns]


Кореляція (Використання vs Сон): -0.79
Кореляція (Залежність vs Психічне здоров'я): -0.95

Порівняння середніх показників за рівнями залежності:


Sleep_Hours_Per_Night  ...  Avg_Daily_Usage_Hours
Addiction_Level                         ...                       
High                          5.728643  ...               6.256784
Low                           8.135294  ...               2.888235
Medium                        7.288957  ...               4.444785

[3 rows x 3 columns]

Знак кореляції: Мінус (-) означає, що коли один показник росте, інший падає.

Порівняння груп: Наскільки сильно відрізняється середній час сну у групи "Low" порівняно з "High"? Це буде ключовим інсайтом для вашого Storytelling.


**Висновки до Блоку 3:**
- _Гіпотеза 1 підтверджена_: Кореляція -0.79 між часом використання та сном свідчить про сильний негативний зв'язок. Студенти, які проводять більше часу в соцмережах, жертвують сном. У групи з високою залежністю сон становить у середньому лише 5.7 годин, що значно нижче норми.
- _Гіпотеза 2 підтверджена_ (екстремальний зв'язок): Кореляція -0.95 між залежністю та психічним здоров'ям є майже лінійною. Це означає, що рівень залежності є ключовим предиктором (показником) психологічного стану студента в цьому наборі даних.
- _Додатковий інсайт:_ Також спостерігається дуже висока кореляція залежності з конфліктами у стосунках (0.93) та впливом на навчання (0.87). Це свідчить про те, що проблема є комплексною і зачіпає всі сфери життя.

**Блок 4: Аналіз платформ, соціальних факторів та успішності (Гіпотези 3, 4, 5 та 6)**
Що ми робимо:
- Аналіз платформ (Гіпотеза 3): Розраховуємо середній рівень залежності для кожної платформи.
- Конфлікти у стосунках (Гіпотеза 4): Порівнюємо середню кількість конфліктів у стосунках через соц.мережі для різних сімейних статусів, щоб перевірити, чи є статус "Complicated" найбільш проблемним.
- Вплив на навчання (Гіпотеза 5): Використовуємо нашу категорію Addiction_Level == 'High', щоб визначити відсоток студентів, які відчувають негативний вплив на навчання.
- Стосунки як захист (Гіпотеза 6): Аналізуємо середній бал залежності (Addicted_Score) для кожного статусу стосунків, щоб побачити, чи дійсно стабільні стосунки ("In Relationship") знижують ризик залежності.

Для чого: Щоб виявити конкретні "групи ризику" та зовнішні чинники (платформи, сімейний стан), які сприяють або, навпаки, запобігають розвитку залежності.

In [4]:
# 1. Аналіз платформ (Гіпотеза 3)
# Групуємо за платформою та рахуємо середній бал залежності
platform_addiction = df_cleaned.groupby('Most_Used_Platform')['Addicted_Score'].mean().sort_values(ascending=False)

print("Середній показник залежності за платформами (Гіпотеза 3):")
print(platform_addiction)

# 2. Аналіз конфліктів залежно від статусу (Гіпотеза 4)
# Перевіряємо, де найбільше конфліктів через соцмережі
relationship_conflicts = df_cleaned.groupby('Relationship_Status')['Conflicts_Over_Social_Media'].mean().sort_values(ascending=False)

print("\nСередня кількість конфліктів за статусом стосунків (Гіпотеза 4):")
print(relationship_conflicts)

# 3. Вплив високої залежності на навчання (Гіпотеза 5)
# Використовуємо створену нами категорію 'High' з Addiction_Level
high_addiction_group = df_cleaned[df_cleaned['Addiction_Level'] == 'High']
academic_impact = high_addiction_group['Affects_Academic_Performance'].value_counts(normalize=True) * 100

print("\nВплив на навчання серед студентів з високим рівнем залежності (Гіпотеза 5):")
if 'Yes' in academic_impact:
    print(f"Відсоток студентів, що відчули вплив: {academic_impact['Yes']:.1f}%")
else:
    print("Дані про негативний вплив відсутні")

# 4. Стосунки як захисний фактор (Гіпотеза 6)
# Порівнюємо рівень залежності між різними статусами стосунків
relationship_protection = df_cleaned.groupby('Relationship_Status')['Addicted_Score'].mean().sort_values()

print("\nСередній бал залежності за статусом стосунків (Гіпотеза 6):")
print(relationship_protection)

Середній показник залежності за платформами (Гіпотеза 3):
Most_Used_Platform
WhatsApp     7.462963
Snapchat     7.461538
TikTok       7.428571
Instagram    6.554217
YouTube      6.100000
WeChat       6.066667
KakaoTalk    6.000000
Facebook     5.666667
Twitter      5.500000
VKontakte    5.000000
LinkedIn     3.809524
LINE         3.000000
Name: Addicted_Score, dtype: float64

Середня кількість конфліктів за статусом стосунків (Гіпотеза 4):
Relationship_Status
Complicated        3.031250
Single             2.901042
In Relationship    2.761246
Name: Conflicts_Over_Social_Media, dtype: float64

Вплив на навчання серед студентів з високим рівнем залежності (Гіпотеза 5):
Відсоток студентів, що відчули вплив: 100.0%

Середній бал залежності за статусом стосунків (Гіпотеза 6):
Relationship_Status
In Relationship    6.342561
Single             6.458333
Complicated        7.031250
Name: Addicted_Score, dtype: float64


**Висновки до Блоку 4:**
- Гіпотеза 3 (Платформи): Підтверджена частково. WhatsApp, Snapchat та TikTok очолюють рейтинг (більше $7.4$). Цікаво, що месенджери з елементами соцмереж (WhatsApp, Snapchat) виявилися навіть вищими за Instagram. LinkedIn очікувано має найнижчий рівень залежності ($3.8$), що підтверджує його професійну спрямованість.
- Гіпотеза 4 (Конфлікти): Підтверджена. Студенти зі статусом "Complicated" мають найбільше конфліктів ($3.03$), тоді як ті, хто "In Relationship", конфліктують найменше ($2.76$).
- Гіпотеза 5 (Навчання): Підтверджена на $100\%$. Кожен студент у вашій вибірці з рівнем залежності "High" відмітив негативний вплив на академічну успішність. Це критичний інсайт.
- Гіпотеза 6 (Захисний фактор): Підтверджена. Студенти у стабільних стосунках мають найнижчий середній бал залежності ($6.34$), тоді як у групі "Complicated" він найвищий ($7.03$). Це доводить, що емоційна стабільність у реальному житті знижує тягу до віртуального.

**Блок 5: Підготовка до Машинного навчання (Кластеризація)**

Тепер спробуємо згрупувати студентів не за однією ознакою, а за сукупністю факторів (час, сон, здоров'я, залежність).

Що ми робимо:
- Масштабування (Scaling): Оскільки різні показники мають різні одиниці (години сну — від 4 до 9, а бал залежності — від 1 до 10), їх треба привести до одного масштабу.
- K-Means Clustering: Автоматично розділимо студентів на 3 логічні групи (кластери).
- Інтерпретація: Подивимось, що це за групи (наприклад: "Свідомі користувачі", "Група ризику", "Залежні").

Для чого: Це дозволить нам не просто аналізувати окремі цифри, а створити "портрети" типових студентів.

In [5]:
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler

# 1. Вибираємо ключові метрики для кластеризації
features_for_clustering = [
    'Avg_Daily_Usage_Hours', 
    'Sleep_Hours_Per_Night', 
    'Mental_Health_Score', 
    'Addicted_Score'
]

x = df_cleaned[features_for_clustering]

# 2. Стандартизація даних (щоб усі ознаки мали однаковий вплив)
scaler = StandardScaler()
x_scaled = scaler.fit_transform(x)

# 3. Використовуємо метод K-Means для створення 3 кластерів
# (Кількість кластерів 3 відповідає нашим рівням Low, Medium, High, але тепер на основі всіх даних)
kmeans = KMeans(n_clusters=3, random_state=42, n_init=10)
df_cleaned['Cluster'] = kmeans.fit_predict(x_scaled)

# 4. Аналіз отриманих кластерів
cluster_summary = df_cleaned.groupby('Cluster')[features_for_clustering].mean()

print("Середні показники для кожного кластера (портрети студентів):")
display(cluster_summary)


# 5. Оновлюємо наш файл, щоб зберегти результати кластеризації
df_cleaned.to_csv('cleaned_social_media_data.csv', index=False)
print("\nДані з кластерами збережено в 'cleaned_social_media_data.csv'")

Середні показники для кожного кластера (портрети студентів):


Avg_Daily_Usage_Hours  ...  Addicted_Score
Cluster                         ...                
0                     3.839858  ...        4.740214
1                     6.475824  ...        8.241758
2                     5.000413  ...        7.049587

[3 rows x 4 columns]


Дані з кластерами збережено в 'cleaned_social_media_data.csv'


**Висновки до Блоку 5:**


Тепер ми маємо не просто цифри, а чіткі профілі студентів. Машинне навчання допомогло нам виділити три різні стратегії поведінки.

Інтерпретація результатів кластеризації:
На основі отриманих середніх значень, ми можемо дати назви цим групам (кластерам) для нашого Storytelling:

**1. Кластер 0 — "Збалансовані" (Balanced Users):**

Характеристики: Найнижче використання (~3.8 год), найкращий сон (~7.8 год) та найвищий рівень психічного здоров'я (~7.4).

Висновок: Це студенти, які використовують соцмережі як інструмент, а не як спосіб життя.

**2. Кластер 2 — "Група ризику" (At-Risk Group):**

Характеристики: Помірне використання (~5 год), сон вже нижче норми (~6.7 год), високий середній бал залежності (~7.0).

Висновок: Ця група вже відчуває негативні наслідки, але ще зберігає відносну стабільність. Саме на них мають бути спрямовані профілактичні заходи.

**3. Кластер 1 — "Цифрова залежність" (Highly Addicted):**

Характеристики: Критично високе використання (~6.5 год), хронічний недосип (~5.6 год) та найнижчі показники здоров'я (~5.0). Найвищий бал залежності (~8.2).

Висновок: Ця група студентів потребує серйозної цифрової детоксикації та, можливо, психологічної допомоги.

**Блок 6: Агрегація даних за географічним принципом (Регіоналізація)**

Мета етапу: Оскільки вихідний набір даних містить велику кількість країн з одиничними відповідями, прямий порівняльний аналіз по країнах є нерепрезентативним. Для отримання статистично значущих результатів ми впроваджуємо групування країн у макрорегіони (континенти).

Завдання:

1. Геокодування: Автоматичне присвоєння кожній країні відповідного континенту за допомогою бібліотеки pycountry_convert з використанням стандартів ISO.

2. Обробка винятків: Створення словника коригування для назв країн, що не відповідають формату ISO (наприклад, США, Великобританія, Ватикан).

3. Збагачення датасету: Створення нової аналітичної ознаки Region, яка стане основою для подальшої візуалізації на карті світу.

In [11]:
# 1. Встановлення бібліотеки (це потрібно зробити один раз)
!pip install pycountry_convert

import pandas as pd
from pycountry_convert import country_name_to_country_alpha2, country_alpha2_to_continent_code

def get_continent(country_name):
    # Пряме виправлення для Ватикану та інших специфічних випадків
    if country_name == 'Vatican City':
        return 'Europe'
    
    try:
        manual_fixes = {
            'USA': 'US', 'UK': 'GB', 'South Korea': 'KR', 'UAE': 'AE', 
            'Russia': 'RU', 'Vietnam': 'VN', 'Czech Republic': 'CZ', 
            'Trinidad': 'TT', 'Kosovo': 'RS', 'Bosnia': 'BA',
            'Bahamas': 'BS', 'Hong Kong': 'HK', 'Taiwan': 'TW'
        }
        
        country_code = manual_fixes.get(country_name)
        if not country_code:
            country_code = country_name_to_country_alpha2(country_name)
            
        continent_code = country_alpha2_to_continent_code(country_code)
        
        continent_map = {
            'AF': 'Africa', 'AS': 'Asia', 'EU': 'Europe', 
            'NA': 'North America', 'SA': 'South America', 'OC': 'Oceania'
        }
        return continent_map.get(continent_code, "Other")
    except:
        return "Other"

# 1. Застосовуємо фінальну функцію
df_cleaned['Region'] = df_cleaned['Country'].apply(get_continent)

# 2. Перевіряємо, чи зникла категорія "Other"
print("Фінальний розподіл за регіонами (100% покриття):")
print(df_cleaned['Region'].value_counts())

# 3. Перевірка Гіпотези 7 з усіма даними
final_region_stats = df_cleaned.groupby('Region')['Addicted_Score'].mean().sort_values(ascending=False)
print("\nРівень залежності за регіонами:")
print(final_region_stats)

# 4. Остаточне збереження очищеного файлу
df_cleaned.to_csv('cleaned_social_media_data.csv', index=False)
print("\nФайл 'cleaned_social_media_data.csv' повністю готовий та збережений!")


[notice] A new release of pip is available: 25.2 -> 25.3
[notice] To update, run: pip install --upgrade pip
Фінальний розподіл за регіонами (100% покриття):
Region
Europe           278
Asia             276
North America    106
Oceania           22
South America     17
Africa             6
Name: count, dtype: int64

Рівень залежності за регіонами:
Region
North America    7.575472
South America    6.764706
Asia             6.673913
Africa           6.166667
Europe           5.827338
Oceania          5.500000
Name: Addicted_Score, dtype: float64

Файл 'cleaned_social_media_data.csv' повністю готовий та збережений!


**Висновок до Блоку 6:**

- Успішність трансформації: Усі 705 записів успішно класифіковані за регіональною ознакою. Це усунуло проблему "малих чисел" і підготувало дані для побудови глобальних дашбордів.
- Перевірка Регіональної гіпотези (№7): Попередній розрахунок підтвердив нерівномірність цифрової залежності. Північна Америка демонструє найвищий рівень залежності ($7.58$), тоді як Океанія та Європа показують помірніші результати ($5.50$ та $5.83$ відповідно).
- Готовність даних: Додавання ознаки Region дозволяє перейти від простої статистики до геопросторового аналізу та сегментації ринку споживання контенту.

## 🎉 Підсумок Етапу 1: Дослідницький аналіз та підготовка даних (DataLab)

У межах першого етапу було успішно реалізовано наступні кроки:

---

### 🧹 Data Cleaning & Quality Control  
*(Очищення та контроль якості даних)*

- Проведено комплексний аудит повноти даних  
- Підтверджено відсутність пропущених значень та дублікатів  
  - **0 пропусків**
  - **0 дублікатів**
- Валідовано типи змінних та перевірено відповідність значень логічним діапазонам  

---

### 🧠 Feature Engineering  
*(Трансформація та створення ознак)*

- Текстові показники (наприклад, `Affects_Academic_Performance`) конвертовано у числовий формат для проведення математичного аналізу та кореляційних розрахунків
- Впроваджено нову категоріальну метрику **`Addiction_Level`**:
  - `Low`
  - `Medium`
  - `High`  
  для якісної оцінки стану респондентів

---

### 📊 Статистична перевірка гіпотез

- Проведено статистичну перевірку **7 гіпотез**
- Кількісно доведено вплив соціальних мереж на ключові сфери життя студентів
- Виявлено **критично високі рівні кореляції** між цифровою залежністю та:
  - порушенням сну
  - погіршенням психічного стану
  - зниженням академічної успішності

---

### 🤖 Сегментація користувачів  
*(Machine Learning)*

- За допомогою алгоритму кластеризації **K-Means** автоматично виділено  
  **три стійкі профілі студентів**:
  - 🟢 **Збалансовані**
  - 🟡 **Група ризику**
  - 🔴 **Цифрова залежність**

---

### 🌍 Геопросторова агрегація  
*(Регіоналізація)*

- Реалізовано групування країн за континентами (регіонами)
- Використано бібліотеку **`pycountry_convert`** з ручним коригуванням винятків
- Це дозволило:
  - усунути *статистичний шум* від країн з поодинокими відповідями
  - підготувати підґрунтя для **макроаналізу**

---

### 💾 Збереження результатів

- Сформовано фінальний збагачений датасет  
  **`cleaned_social_media_data.csv`**
- Датасет виступає **"золотим стандартом"** для:
  - подальшої візуалізації
  - аналітики
  - розробки інтерактивного додатку

---

### ✅ Результат етапу

Дані пройшли **повний цикл обробки**  
*(від Raw Data до Clean Data)*  
та повністю готові до:

- інтеграції в **GitHub**
- деплою у **Streamlit**
